### ISSUE TITLE
actions on plug and play DRL notebook

### ISSUE THREAD

**Issue by wezzyzzs** [AI4Finance-Foundation/FinRL-Meta#15]
Hello, I'm a beginner of RL. 
I wonder that if we can see the actions with the ElegentRL's  DRL_prediction function ? 
(Return actions in [-1,0 1], just like the sb3's DRL_prediction function).
It will be helpful if someone share their idea, thanks!

Besides, I wonder that is it suitable to trade index (like DJI) as single stock with the env in plug_and_play_DRL notebook ?

**Comment by rayrui312** [comment]
1. Yes. You can print out the actions by adding codes between lines 121 and 122 in elegantrl_models.py.
`...`
`                action = (
                    a_tensor.detach().cpu().numpy()[0]` 
`print(action)`
`state, reward, done, _ = environment.step(action)`
`...`

2. Yes. It can work. But you may need to tune the states and the hyperparameters to get a good result.

**Comment by wezzyzzs** [comment]
@rayrui312  Thank for your reply!
But when I add the code you mention , it seems to be weird 

![messageImage_1637250349966](https://user-images.githubusercontent.com/86904909/142448544-07ed100c-7f36-45e2-a470-086bbd5e494e.jpg)

I used AAPL and PPO algorithm to trained the model , but it didn't return actions with [-1,0,1]
Instead, the agent return float just like the picture show.
https://github.com/AI4Finance-Foundation/FinRL/issues/346
I wonder if my model get wrong or I just miss something. Thank you!

**Comment by rayrui312** [comment]
Because you are using the PPO algorithm, the actions are continuous values between -1 to 1. We time it by 100 to represent the number of shares to buy (negative means sell).  The -1, 0, 1 action design is for algorithms that use discrete actions, such as DQN.

**Comment by wezzyzzs** [comment]
@rayrui312 I got it, thanks!
but when I set the initial capital as 200 (agent only can buy 1 share of APPL), the actions still showed just like this
![圖片](https://user-images.githubusercontent.com/86904909/142720680-75da0abb-83ba-4609-b835-2bd48edf16c2.png)


However, when I checked the account_value_erl, the balance account showed  just bought 1 share at day 0 and held it until the last day.
![圖片](https://user-images.githubusercontent.com/86904909/142720687-db870681-cfa7-43ab-82a7-36d4cc42a927.png)

So I wonder in this case, how can we explain the actions that agent returns, thank for your help!

**Comment by rayrui312** [comment]
If the cash is not enough, we will use all the money to buy the shares. So you can observe this phenomenon.

**Comment by wezzyzzs** [comment]
@rayrui312 
I thought when cash is not enough, the agent will return 0 as the action.
Just like the AAPL case, the agent buy 1 share in day 2, the account will be 199.89.
And day 3 , the agent hold 1 share AAPL, but the action is still 0.4963. I thought it will return 0 because there is not trade happened

I am confuseed by the result of actions. Does it mean it's not suitable to restrain the agent to buy one sell one share?

**Comment by rayrui312** [comment]
The agent action doesn't mean the actual trade. For this env, it's not suitable to restrain the agent to buy one sell one share. You may use the DQN algorithm with discrete actions if you want the agent to buy/sell one share at each step.

**Comment by zhumingpassional** [comment]
> @rayrui312 I thought when cash is not enough, the agent will return 0 for the action. Just like the AAPL case, the agent buy 1 share in day 2, the account will be 199.89. And day 3 , the agent hold 1 share AAPL, but the action is still 0.4963. I thought it will return 0 because there is not trade happened
> 
> I am confuseed by the result of actions. Does it mean it's not suitable to restrain the agent to buy one sell one share?

It (e.g., [0,43],[0.49]) means that in the next day, the first
